pallets/parachain-staking/src/lib.rs

```diff
@@ -2550,7 +2606,20 @@ pub mod pallet {
 
 	impl<T: Config> ShouldEndSession<T::BlockNumber> for Pallet<T> {
 		fn should_end_session(now: T::BlockNumber) -> bool {
-			<Round<T>>::get().should_update(now)
+			let mut round = <Round<T>>::get();
+			// always update when a new round should start
+			if round.should_update(now) {
+				true
+			} else if <ForceNewRound<T>>::get() {
```

Comment:
Maybe we should register extra weight unchecked here?
```rust
frame_system::Pallet::<T>::register_extra_weight_unchecked(
  T::DbWeight::get().reads_writes(1, 2),
  DispatchClass::Mandatory,
);
```

Reply:
yes we should!

Reply:
Why is it 1 read only? Don't we also always read from `Round<T>`? Does that come "for free"?

Reply:
This was exemplary. See ac6a011.